### PR TITLE
Update create-map.ts

### PR DESCRIPTION
**Pull request by juandjara**

### CODE REVIEW

File: src/lib/viz/basemap/create-map.ts [hunk 1]
```diff
@@ -55,9 +55,12 @@ export function createMap(options: DeckGLMapOptions = DEFAULT_OPTIONS) {
 
   const deckMap = new (window.deck.DeckGL as any)({
     mapStyle: CartoMapStyle[chosenOptions.basemap.toUpperCase() as keyof typeof CartoMapStyle],
-    initialViewState: chosenOptions.view,
+    viewState: chosenOptions.view,
     container: chosenOptions.container,
-    controller: true
+    controller: true,
+    onViewStateChange: ({ viewState }) => {
+      deckMap.setProps({ viewState })
```
Comment (jesusbotella) [inline src/lib/viz/basemap/create-map.ts]: Not sure about this. We'd be supplying the very same viewState to the map whenever it changes. Seems like an unnecessary overhead.

**Comment by VictorVelarde** [comment]
From the docs: https://deck.gl/#/documentation/developer-guide/views-and-projections?section=using-a-view-class

If **initialViewState** is provided, deck.gl automatically tracks the view states of interactive views (used as a "stateful" component):

If you need to manage and manipulate the view state outside of deck.gl, you may do so by providing an external **viewState** prop (used as a "stateless" component). In this case, you also need to listen to the onViewStateChange callback and update the viewState object yourself:

**Comment by juandjara** [comment]
@VictorVelarde thanks for the link. That describes exactly what we want to do, "manage and manipulate the view state outside of deck.gl"

File: src/lib/viz/basemap/create-map.ts (hunk 1 above)
Comment (simon-contreras-deel) [reply src/lib/viz/basemap/create-map.ts]: I agree

Comment (simon-contreras-deel) [reply src/lib/viz/basemap/create-map.ts]: I think we would be forcing it to change state when "the state has changed". I mean, creating a never-ending state update loop, don't you think?

Comment (juandjara) [reply src/lib/viz/basemap/create-map.ts]: As it says in the comment from Victor below, the map does not update itself with pan and zoom events if `initialViewState` is not set. What it does is just send a `onViewStateChange` event with the new viewState so you can update it with `setProps`

**Comment by sclebal** [comment]
@VictorVelarde @jesusbotella @oleurud Without this change when I change the view the map remain static. This PR is necessary to avoid it. Thanks @juandjara.

**Comment by VictorVelarde** [comment]
I'm working on this

**Review by VictorVelarde — APPROVED** [review]
I have made just small adjustments, and I added a debug example to test fit-bounds

**Pull request merged**